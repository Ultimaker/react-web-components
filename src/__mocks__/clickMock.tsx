// Copyright (c) 2018 Ultimaker B.V.

export const mockClickEvent = { stopPropagation: jest.fn(), preventDefault: jest.fn() };
