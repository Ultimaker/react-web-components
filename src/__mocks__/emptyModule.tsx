// Copyright (c) 2018 Ultimaker B.V.
module.exports = {};
