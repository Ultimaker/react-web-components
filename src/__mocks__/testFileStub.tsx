// Copyright (c) 2018 Ultimaker B.V.
module.exports = 'test-file-stub';