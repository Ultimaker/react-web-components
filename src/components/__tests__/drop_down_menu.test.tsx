// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow } from 'enzyme';

// component
import DropDownMenu from '../drop_down_menu';

// mocks
import { mockClickEvent } from '../../__mocks__/clickMock';


describe('The DropDownMenu component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            activeLabel: 'Active label'
        };
        wrapper = shallow(<DropDownMenu {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should show menu when the label is clicked', () => {
        wrapper.find('.label').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible')).toHaveLength(1);
    });

    it('should hide menu when the label is clicked', () => {
        wrapper.find('.label').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible')).toHaveLength(1);
        wrapper.find('.label').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible').exists()).toEqual(false);
    });

    it('should hide menu on blur', () => {
        wrapper.find('.label').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible')).toHaveLength(1);
        wrapper.simulate('blur');
        expect(wrapper.find('.visible').exists()).toEqual(false);
    });

    it('should hide menu on menu click', () => {
        wrapper.find('.label').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible')).toHaveLength(1);
        wrapper.find('.container').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible').exists()).toEqual(false);
    });

    it('should not propagate click', () => {
        wrapper.simulate('click', mockClickEvent);
        expect(mockClickEvent.stopPropagation).toBeCalled();
    });

});
