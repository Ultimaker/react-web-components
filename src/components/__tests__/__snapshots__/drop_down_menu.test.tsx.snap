// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DropDownMenu component should render 1`] = `
<div
  className="drop-down-menu"
>
  <DropDownMenuBase
    onToggleMenuHandler={[Function]}
    showMenu={false}
    triggerElement={
      <div
        className="layout layout--align-middle layout--gutter-none"
      >
        <div
          className="layout__item u-fit"
        >
          <div
            className="drop-down-menu-base__button-label"
          >
            Active label
          </div>
        </div>
        <div
          className="layout__item u-fit layout__item--right"
        >
          <PanelArrow
            active={false}
            color="blue"
            width="1.2rem"
          />
        </div>
      </div>
    }
  />
</div>
`;
